lib/stateful_enum/state_inspection.rb

```diff
@@ -13,18 +13,28 @@ def stateful_enum
     end
 
     def stateful_enum
-      StateInspector.new(self.class.stateful_enum, self)
+      self.class.stateful_enum.map do |column, defined_stateful_enum|
+        StateInspector.new(defined_stateful_enum, self)
```

Comment:
I'm proposing to make `StateInspector` to work only with one `defined_stateful_enum` instance at a time, that'll simplify the logic inside of it